src/Microsoft.OData.Edm/Csdl/Parsing/SchemaJsonParser.cs

```diff
@@ -0,0 +1,1564 @@
+                        }
+                    }
+                }
+
+                prameterAnnotations.Remove(optionalAnnotationValue);
+            }
+
+            CsdlOperationParameter csdlParameter = new CsdlOperationParameter(name, parameterType, context.Location(), isOptional, defaultValue);
+            prameterAnnotations.ForEach(v => csdlParameter.AddAnnotation(v));
+
+            return csdlParameter;
+        }
+
+        /// <summary>
+        /// The value of $ReturnType is an object.
+        /// </summary>
+        /// <param name="name"></param>
+        /// <param name="jsonValue"></param>
+        /// <param name="jsonPath"></param>
+        /// <returns></returns>
+        internal static CsdlOperationReturn ParseCsdlOperationReturn(string name, JsonElement element, JsonParserContext context)
+        {
+            // The value of $ReturnType is an object.
+            Debug.Assert(element.ValueKind == JsonValueKind.Object);
+            Debug.Assert(context != null);
+            Debug.Assert(name == "$ReturnType", "The name should be $ReturnType.");
+
+            // It MAY contain the members $Type, $Collection, $Nullable, $MaxLength, $Unicode, $Precision, $Scale, and $SRID.
+            CsdlTypeReference typeReference = CsdlJsonParseHelper.ParseCsdlTypeReference(element, context);
+
+            IList<CsdlAnnotation> annotations = new List<CsdlAnnotation>();
+            element.ParseAsObject(context, (propertyName, propertyValue) =>
+            {
+                switch (propertyName)
+                {
+                    case "$Type":
+                    case "$Collection":
+                    case "$Nullable":
+                    case "$MaxLength":
+                    case "$Unicode":
+                    case "$Precision":
+                    case "$Scale":
+                    case "$SRID":
+                        // skip them because verified or not supported.
+                        break;
+
+                    default:
+                        // it MAY contain annotations.
+                        ParseCsdlAnnotation(propertyName, propertyValue, context, annotations);
+                        break;
+                }
+            });
+
+            CsdlOperationReturn returnType = new CsdlOperationReturn(typeReference, context.Location());
+            annotations.ForEach(a => returnType.AddAnnotation(a));
+            return returnType;
+        }
+
+        #endregion
+
+        /// <summary>
+        /// The schema object MAY contain members representing:
+        /// entity types, complex types, enumeration types, type definitions, actions, functions, terms, and an entity container.
+        /// This method trys to build the JSON value as one of the above member.
+        /// </summary>
+        /// <param name="name">The name of the schema member.</param>
+        /// <param name="element">The schema member json value.</param>
+        /// <param name="jsonPath">The JSON path.</param>

```

Comment:
fix here (and in other places) to match function signature

Reply:
Updated, also for others in this class.

---
In reply to: [443699157](https://github.com/OData/odata.net/pull/1775#discussion_r443699157) [](ancestors = 443699157)